range.c

```diff
@@ -1888,6 +1888,49 @@ static int r_cover_range_p(VALUE range, VALUE beg, VALUE end, VALUE val);
  *  - An internal call to <tt><=></tt> returns +nil+;
  *    that is, the operands are not comparable.
  *
+ * Beginless ranges cover all values of the same type before the end,
+ * excluding the end for exclusive ranges. Beginless ranges cover
+ * ranges that end before the end of the beginless range, or at the
+ * end of the beginless range for inclusive ranges.
+ *
+ *    (..2).cover?(1)     # => true
+ *    (..2).cover?(2)     # => true
+ *    (..2).cover?(3)     # => false
+ *    (...2).cover?(2)    # => false
+ *    (..2).cover?("2")   # => false
+ *    (..2).cover?(..2)   # => true
+ *    (..2).cover?(...2)  # => true
+ *    (..2).cover?(.."2") # => false
+ *    (...2).cover?(..2)  # => false
+ *
+ * Endless ranges cover all values of the same type after the
+ * beginning. Endless exclusive ranges do not cover endless
+ * inclusive ranges.
+ *
+ *    (2..).cover?(1)     # => false
+ *    (2..).cover?(3)     # => true
+ *    (2...).cover?(3)    # => true
+ *    (2..).cover?(2)     # => true
+ *    (2..).cover?("2")   # => false
+ *    (2..).cover?(2..)   # => true
+ *    (2..).cover?(2...)  # => true
+ *    (2..).cover?("2"..) # => false
+ *    (2...).cover?(2..)  # => false
+ *    (2...).cover?(3...) # => true
+ *    (2...).cover?(3..)  # => false
+ *    (3..).cover?(2..)   # => false
+ *
+ * Ranges that are both beginless and endless cover all values and
+ * ranges, and return true for all arguments, with the exception that
+ * beginless and endless exclusive ranges do not cover endless
+ * inclusive ranges.
+ *
+ *    (nil...).cover?(Object.new) # => true
```

Comment:
Although beginless-endless ranges can be written as `(nil..)` or `(nil...)`, I think using that in the example would cause unwanted confusion because of the asymmetry of omission. I think it would be clearer to use `nil..nil` and `nil...nil`, or at least remind the readers that `nil` on one side (but not both sides) can be omitted, so that `(nil..)`, `(..nil)`, and `nil..nil` all mean the same thing.

Reply:
`Range#inspect` uses `nil..`, not `nil..nil`, indicating to me that `(nil..)` is the preferred form.